### PR TITLE
refactoring: auth domain dependency set

### PR DESCRIPTION
**Pull request by Migorithm**
### Issue Link

-

### Description

-

### Changes

- dependency set for auth domain

### CODE REVIEW

File: src/domains/auth/mod.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+#[derive(Serialize)]
+pub enum AccountStatus {
+    Active,
+    Deleted,
+    Abnormal,
+}
+
+#[derive(Serialize)]
+pub struct Account {
```
Comment (seonwoo960000) [inline src/domains/auth/mod.rs:17]: What do you think of naming the package `account`

Comment (Migorithm) [reply src/domains/auth/mod.rs:17]: @seonwoo960000 no problemo

File: src/services/account/repository.rs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+use crate::domains::account::Account;
+
+#[async_trait]
+pub trait AccountRepository {
+    async fn get(&self, id: i64) -> Account;
+
+    async fn add(&self, account: &Account);
+
+    async fn update(&self, account: &Account);
```
Comment (seonwoo960000) [inline src/services/account/repository.rs:11]: I wonder if there is a way we can know the result of `add` and `update` operation. If `add` and `update` fails for any reason, how can we know whether it failed or succeeded? Is it possible to use `Result` here? Btw sorry for my lack of understanding of Rust.

**Comment by seonwoo960000** [comment]
I wonder how we can pass `Pool<Postgres>`(connection pool to run sql queries) to the repository layer. Should we create the `Pool<Postgres>`in the `main` function and pass it all the way down to repository? 
- `main` -> `router` -> `service` -> `repository` ?

**Pull request merged**